### PR TITLE
feat(formatting): add network throughput number formats for bits/s and bytes/s

### PR DESCRIPTION
**Pull request by bmuraykhi**
### SUMMARY

Adds two decimal (base-1000) number formats for network throughput, rendered on the bit
scale `bps` / `kbps` / `Mbps` / `Gbps` / `Tbps`:

| Format ID                       | Label                                                    | Column holds          |
| ------------------------------- | -------------------------------------------------------- | --------------------- |
| `NETWORK_THROUGHPUT`            | Network throughput in bits/s (1000000 => 1Mbps)          | bits per second       |
| `NETWORK_THROUGHPUT_FROM_BYTES` | Network throughput in bytes/s as bits (1000000 => 8Mbps) | bytes per second (×8) |

**Why:** `MEMORY_TRANSFER_RATE_DECIMAL` already renders a byte rate _as bytes_
(`1024B => 1.024kB/s`), but no existing option renders a rate _as bits_ — even though
network throughput is conventionally reported that way. Today a column already reporting
bits/s has no matching format at all, and a byte-rate column needs `× 8` written into every
metric expression to display Mbps.

**Why two entries:** they differ only in the unit of the source column, and the same input
makes that visible — `1000000` renders as `1Mbps` under the first and `8Mbps` under the
second. Collapsing them into one would force every user to know which convention was
assumed.

**Implementation:** a new `createThroughputFormatter` factory in `@superset-ui/core`,
taking a `fromBytes` flag — mirroring the flag style of the neighbouring
`createMemoryFormatter({ binary, transfer })`. The unmarked default takes the natural unit
(bits/s) and only the converting variant carries a suffix, following the `LENGTH` /
`LENGTH_CM_KM` / `LENGTH_CM_M` precedent in the same file.

**No decimal/binary split:** network rates are universally SI, so a binary counterpart
would disambiguate nothing. This matches `DURATION` and `LENGTH`, which likewise carry no
such marker; only `MEMORY_*` does, because kB and KiB genuinely both exist.

The two entries sit alongside the other rate formats, directly after
`Memory transfer rate in bytes - decimal`.

Requested in https://github.com/apache/superset/discussions/42436.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

**Before** — Customize → Number format, with no throughput options available:

before

**After** — `Network throughput in bits/s`, where `1000000` renders as `1Mbps`:

after-bits

**After** — `Network throughput in bytes/s as bits`, where the same `1000000` renders as `8Mbps`:

after-bytes

### TESTING INSTRUCTIONS

Manual:

1. Open any chart in Explore and go to the **Customize** tab.
2. Open **Number format**. Two new entries appear next to the existing memory transfer rate
   options:
   - `Network throughput in bits/s (1000000 => 1Mbps)`
   - `Network throughput in bytes/s as bits (1000000 => 8Mbps)`
3. Select the first against a numeric column — a value of `1000000` renders as `1Mbps`, and
   `1500000` as `1.5Mbps`.
4. Select the second — the same `1000000` renders as `8Mbps`, and `125000000` as `1Gbps`.

Unit tests:

```bash
cd superset-frontend
npm run test -- createThroughputFormatter
```

12 tests, 100% statement/branch/function/line coverage on the new factory. They cover both
conversion paths, the full suffix range through `Qbps`, clamping above and below the known
range, promotion to the next suffix when rounding reaches the base (`999999 => 1Mbps`),
negative rates, and the `decimals` option.

`src/setup/setupFormatters.test.ts` additionally asserts that both format IDs are registered
against the correct converter, so a dropped, renamed, or swapped registration fails.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/discussions/42436 (feature request, moved to Discussions per the contributing guide)
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/packages/superset-ui-core/src/number-format/factories/createThroughputFormatter.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    const magnitude = Math.abs(value);
+    const bits = fromBytes ? magnitude * BITS_PER_BYTE : magnitude;
+    const i = Math.max(
+      0,
+      Math.min(
+        SUFFIXES.length - 1,
+        Math.floor(Math.log(bits) / Math.log(BASE)),
+      ),
+    );
+    const scaled = parseFloat((bits / Math.pow(BASE, i)).toFixed(decimals));
```
[the remark this answers is not in the document]
Comment (bmuraykhi) [reply superset-frontend/packages/superset-ui-core/src/number-format/factories/createThroughputFormatter.ts]: fixed in 07e98d2551. The suffix index is now re-evaluated after rounding, guarded so the largest suffix still saturates rather than overflowing (1000Qbps is preserved). Affected inputs and verification details are in the PR comment below.

**Comment by bmuraykhi** [comment]
Both review findings are addressed.

**Rounding at unit boundaries** (07e98d2551). The suffix index was taken from the raw value *before* rounding, so anything that rounded up to the base kept the previous unit:

| input | before | expected |
|---|---|---|
| `999999` | `1000kbps` | `1Mbps` |
| `999999999` | `1000Mbps` | `1Gbps` |
| `999.999` | `1000bps` | `1kbps` |
| `999.6` (`decimals: 0`) | `1000bps` | `1kbps` |
| `124999.99` (`fromBytes`) | `1000kbps` | `1Mbps` |

The fix re-evaluates the index after rounding, guarded with `i < SUFFIXES.length - 1` so the largest suffix saturates rather than overflowing — `1000Qbps` is preserved. Every pre-existing assertion produces identical output, plus a new test covering both signs, both conversion paths, and `decimals: 0`.

**Missing assertions for the registrations** (3aacc27e92). `setupFormatters.test.ts` now asserts both `NETWORK_THROUGHPUT` and `NETWORK_THROUGHPUT_FROM_BYTES` are registered against the formatter instance returned by the matching `createThroughputFormatter` call, so a dropped, renamed, or swapped registration fails.

**One pre-existing bug worth flagging.** While confirming the rounding fix I found that `createMemoryFormatter` has the same bug, plus a second one at the bottom of the scale:

```js
createMemoryFormatter()(999999)  // '1000kB'       — expected '1MB'
createMemoryFormatter()(0.5)     // '500undefined' — index reaches -1, so the value is
                                 //   divided by 1000⁻¹ and suffixes[-1] is undefined
```

`createThroughputFormatter` guards against both — the index is clamped with `Math.max(0, …)` and re-evaluated after rounding — which is why the new code looks slightly different from its neighbour. I've left `createMemoryFormatter` untouched to keep this PR scoped to the new formats, and will open a separate `fix:` PR for it. Happy to fold it in here instead if you'd prefer it as one change.

**Review by rusackas — APPROVED** [review]
Nice, self-contained addition. Formatter's isolated, tests cover the boundary cases well, and the rounding bug CodeAnt flagged got fixed fast with good verification. LGTM.
